components/button/button.tsx

```diff
@@ -262,7 +267,7 @@ const InternalCompoundedButton = React.forwardRef<
           : (e as React.MouseEvent<HTMLButtonElement, MouseEvent>),
       );
     },
-    [props.onClick, innerLoading, mergedDisabled],
+    [props.onClick, mergedDisabled],
```

Comment:
Since we still use `innerLoading.current` as state usage, you can replace useCallback with `useEvent` from rc util instead. Which will no need deps anymore.